cmd/copyUtil.go

```diff
@@ -157,19 +165,15 @@ func (util copyHandlerUtil) urlIsAzureFileDirectory(ctx context.Context, url *ur
 	return true
 }
 
-func (util copyHandlerUtil) getContainerUrl(blobParts azblob.BlobURLParts) url.URL {
-	blobParts.BlobName = ""
-	return blobParts.URL()
-}
-
 // doesBlobRepresentAFolder verifies whether blob is valid or not.
 // Used to handle special scenarios or conditions.
-func (util copyHandlerUtil) doesBlobRepresentAFolder(metadata azblob.Metadata) bool {
+func (util copyHandlerUtil) doesBlobRepresentAFolder(metadata map[string]*string) bool {
 	// this condition is to handle the WASB V1 directory structure.
 	// HDFS driver creates a blob for the empty directories (let’s call it ‘myfolder’)
 	// and names all the blobs under ‘myfolder’ as such: ‘myfolder/myblob’
 	// The empty directory has meta-data 'hdi_isfolder = true'
-	return metadata["hdi_isfolder"] == "true"
+	// Note: GoLang sometimes sets metadata keys with the first letter capitalized
+	return (metadata["hdi_isfolder"] != nil && strings.ToLower(*metadata["hdi_isfolder"]) == "true") || (metadata["Hdi_isfolder"] != nil && strings.ToLower(*metadata["Hdi_isfolder"]) == "true")
```

Comment:
in track-2 while setting the metadata we will always preserve the case user has given but reading is still messy I guess.

Reply:
This might be due to intermingling Track 1 and Track 2 with tests. Should we keep this anyway since a customer may have uploaded with Track 1.5?

Reply:
This is _ok_, but a little ugly, but I still prefer it over the alternative (enumerating over & EqualFolding the keys).